File: constants/tests/resultdatasourcescompare.cs
/**
  This is the definition of the Data Sources supported by Capture

  2018 © Socket Mobile, Inc. all rights reserved
*/

using System;

namespace SocketMobile
{
    namespace Capture
    {
        /// <summary>
        /// defines a symbology
        /// </summary>
        public abstract class ICaptureSymbology
        {
            /// <summary>
            /// ID of the symbology
            /// </summary>
            public abstract int ID { get; set; }
            /// <summary>
            /// Flags to know if the Param property is used
            /// </summary>
            public abstract int Flags { get; set; }
            /// <summary>
            /// Current status of a symbology
            /// </summary>
            public abstract int Status { get; set; }
            /// <summary>
            /// Parameters of a symbology (reserved for future use)
            /// </summary>
            public abstract int Param { get; set; }
            /// <summary>
            /// Name of the Symbology
            /// </summary>
            public abstract String Name { get; }

            /// <summary>
/// defines the Data Source IDs used in Capture
/// </summary>
public class Id
{
	/// <summary>
	/// the data source ID is not specified or initialized
	/// </summary>
	public const int kNotSpecified = 0;

	/// <summary>
	/// the barcode symbology for Australia post
	/// </summary>
	public const int kSymbologyAustraliaPost = 1;

	/// <summary>
	/// the barcode symbology Aztec
	/// </summary>
	public const int kSymbologyAztec = 2;

	/// <summary>
	/// the barcode symbology Bookland EAN
	/// </summary>
	public const int kSymbologyBooklandEan = 3;

	/// <summary>
	/// the barcode symbology for British post
	/// </summary>
	public const int kSymbologyBritishPost = 4;

	/// <summary>
	/// the barcode symbology for Canada post
	/// </summary>
	public const int kSymbologyCanadaPost = 5;

	/// <summary>
	/// the barcode symbology Chinese 2 of 5
	/// </summary>
	public const int kSymbologyChinese2of5 = 6;

	/// <summary>
	/// the barcode symbology Codabar
	/// </summary>
	public const int kSymbologyCodabar = 7;

	/// <summary>
	/// the barcode symbology Codablock A
	/// </summary>
	public const int kSymbologyCodablockA = 8;

	/// <summary>
	/// the barcode symbology Codablock F
	/// </summary>
	public const int kSymbologyCodablockF = 9;

	/// <summary>
	/// the barcode symbology Code 11
	/// </summary>
	public const int kSymbologyCode11 = 10;

	/// <summary>
	/// the barcode symbology Code 39
	/// </summary>
	public const int kSymbologyCode39 = 11;

	/// <summary>
	/// the barcode symbology Code 39 Extended
	/// </summary>
	public const int kSymbologyCode39Extended = 12;

	/// <summary>
	/// the barcode symbology Code 39 Trioptic
	/// </summary>
	public const int kSymbologyCode39Trioptic = 13;

	/// <summary>
	/// the barcode symbology Code 93
	/// </summary>
	public const int kSymbologyCode93 = 14;

	/// <summary>
	/// the barcode symbology Code 128
	/// </summary>
	public const int kSymbologyCode128 = 15;

	/// <summary>
	/// the barcode symbology DataMatrix
	/// </summary>
	public const int kSymbologyDataMatrix = 16;

	/// <summary>
	/// the barcode symbology for Dutch post
	/// </summary>
	public const int kSymbologyDutchPost = 17;

	/// <summary>
	/// the barcode symbology EAN 8
	/// </summary>
	public const int kSymbologyEan8 = 18;

	/// <summary>
	/// the barcode symbology EAN 13
	/// </summary>
	public const int kSymbologyEan13 = 19;

	/// <summary>
	/// the barcode symbology EAN 128
	/// </summary>
	public const int kSymbologyEan128 = 20;

	/// <summary>
	/// the barcode symbology EAN 128 Irregular
	/// </summary>
	public const int kSymbologyEan128Irregular = 21;

	/// <summary>
	/// the barcode symbology EAN UCC Composite AB
	/// </summary>
	public const int kSymbologyEanUccCompositeAB = 22;

	/// <summary>
	/// the barcode symbology EAN UCC Composite C
	/// </summary>
	public const int kSymbologyEanUccCompositeC = 23;

	/// <summary>
	/// the barcode symbology GS1 Databar
	/// </summary>
	public const int kSymbologyGs1Databar = 24;

	/// <summary>
	/// the barcode symbology GS1 Databar Limited
	/// </summary>
	public const int kSymbologyGs1DatabarLimited = 25;

	/// <summary>
	/// the barcode symbology GS1 Databar Expanded
	/// </summary>
	public const int kSymbologyGs1DatabarExpanded = 26;

	/// <summary>
	/// the barcode symbology Interleaved 2 of 5
	/// </summary>
	public const int kSymbologyInterleaved2of5 = 27;

	/// <summary>
	/// the barcode symbology ISBT 128
	/// </summary>
	public const int kSymbologyIsbt128 = 28;

	/// <summary>
	/// the barcode symbology for Japan post
	/// </summary>
	public const int kSymbologyJapanPost = 29;

	/// <summary>
	/// the barcode symbology Matrix 2 of 5
	/// </summary>
	public const int kSymbologyMatrix2of5 = 30;

	/// <summary>
	/// the barcode symbology Maxi Code
	/// </summary>
	public const int kSymbologyMaxicode = 31;

	/// <summary>
	/// the barcode symbology MSI
	/// </summary>
	public const int kSymbologyMsi = 32;

	/// <summary>
	/// the barcode symbology PDF 417
	/// </summary>
	public const int kSymbologyPdf417 = 33;

	/// <summary>
	/// the barcode symbology PDF 417 Micro
	/// </summary>
	public const int kSymbologyPdf417Micro = 34;

	/// <summary>
	/// the barcode symbology Planet
	/// </summary>
	public const int kSymbologyPlanet = 35;

	/// <summary>
	/// the barcode symbology Plessey
	/// </summary>
	public const int kSymbologyPlessey = 36;

	/// <summary>
	/// the barcode symbology Postnet
	/// </summary>
	public const int kSymbologyPostnet = 37;

	/// <summary>
	/// the barcode symbology QR Code
	/// </summary>
	public const int kSymbologyQRCode = 38;

	/// <summary>
	/// the barcode symbology Standard 2 of 5
	/// </summary>
	public const int kSymbologyStandard2of5 = 39;

	/// <summary>
	/// the barcode symbology Telepen
	/// </summary>
	public const int kSymbologyTelepen = 40;

	/// <summary>
	/// the barcode symbology TLC 39
	/// </summary>
	public const int kSymbologyTlc39 = 41;

	/// <summary>
	/// the barcode symbology UPC A
	/// </summary>
	public const int kSymbologyUpcA = 42;

	/// <summary>
	/// the barcode symbology UPC E0
	/// </summary>
	public const int kSymbologyUpcE0 = 43;

	/// <summary>
	/// the barcode symbology UPC E1
	/// </summary>
	public const int kSymbologyUpcE1 = 44;

	/// <summary>
	/// the barcode symbology USPS Intelligent Mail
	/// </summary>
	public const int kSymbologyUspsIntelligentMail = 45;

	/// <summary>
	/// the barcode symbology Direct Part Marking
	/// </summary>
	public const int kSymbologyDirectPartMarking = 46;

	/// <summary>
	/// the barcode symbology Han Xin
	/// </summary>
	public const int kSymbologyHanXin = 47;

	/// <summary>
	/// the barcode symbology DotCode
	/// </summary>
	public const int kSymbologyDotCode = 48;

	/// <summary>
	/// the barcode symbology Digimarc
	/// </summary>
	public const int kSymbologyDigimarc = 49;

	/// <summary>
	/// the last barcode symbology ID, not an actual barcode symbology
	/// </summary>
	public const int kLastSymbologyID = 50;

	/// <summary>
	/// the RFID Tag Type ISO 14443 A
	/// </summary>
	public const int kTagTypeISO14443TypeA = 256;

	/// <summary>
	/// the RFID Tag Type ISO 14443 B
	/// </summary>
	public const int kTagTypeISO14443TypeB = 257;

	/// <summary>
	/// the RFID Tag Type Felica
	/// </summary>
	public const int kTagTypeFelica = 258;

	/// <summary>
	/// the RFID Tag Type ISO 15693
	/// </summary>
	public const int kTagTypeISO15693 = 259;

	/// <summary>
	/// the RFID Tag Type NXPI Code 1
	/// </summary>
	public const int kTagTypeNXPICODE1 = 260;

	/// <summary>
	/// the RFID Tag Type Inside Secure Pico Tag
	/// </summary>
	public const int kTagTypeInsideSecurePicoTag = 261;

	/// <summary>
	/// the RFID Tag Type Innovision Topaz Jewel
	/// </summary>
	public const int kTagTypeInnovisionTopazJewel = 262;

	/// <summary>
	/// the RFID Tag Type Thin Film NFC Barcode
	/// </summary>
	public const int kTagTypeThinfilmNFCBarcode = 263;

	/// <summary>
	/// the RFID Tag Type ST Micro Electronics SR
	/// </summary>
	public const int kTagTypeSTMicroElectronicsSR = 264;

	/// <summary>
	/// the RFID Tag Type ASK CTS 256B or CTS 512B
	/// </summary>
	public const int kTagTypeASKCTS256BOrCTS512B = 265;

	/// <summary>
	/// the RFID Tag Type NFC Forum
	/// </summary>
	public const int kTagTypeNFCForum = 266;

	/// <summary>
	/// the RFID Tag Type Innovatron Radio Protocol
	/// </summary>
	public const int kTagTypeInnovatronRadioProtocol = 267;

	/// <summary>
	/// the last RFID tag type, not an actual tag type
	/// </summary>
	public const int kTagTypeLastTagType = 268;

};



            /// <summary>
/// defines the Data Source Flags used in a Capture Data Source
/// </summary>
public class FlagsMask
{
	/// <summary>
	/// the data source contains a status
	/// </summary>
	public const int kStatus = 1;

	/// <summary>
	/// the data source contains some parameters
	/// </summary>
	public const int kParam = 2;

};



            /// <summary>
/// defines the Data Source Status used in Capture
/// </summary>
public class StatusValues
{
	/// <summary>
	/// the data source status is disabled
	/// </summary>
	public const int kDisable = 0;

	/// <summary>
	/// the data source status is enabled
	/// </summary>
	public const int kEnable = 1;

	/// <summary>
	/// the data source is not supported
	/// </summary>
	public const int kNotSupported = 2;

};


          }
      }
  }
  

File: constants/tests/resultdatasourcescompare.ts
/**
  @file CaptureDataSource.ts
  @brief Define the Data Source object in Capture

  @copyright © 2022 Socket Mobile, Inc.

 */

/**

 @brief idenfity a barcode symbology or a RFID/NFC Tag

 */
export class CaptureDataSourceID {
	// the data source ID is not specified or initialized
	static readonly NotSpecified = 0;

	// the barcode symbology for Australia post
	static readonly SymbologyAustraliaPost = 1;

	// the barcode symbology Aztec
	static readonly SymbologyAztec = 2;

	// the barcode symbology Bookland EAN
	static readonly SymbologyBooklandEan = 3;

	// the barcode symbology for British post
	static readonly SymbologyBritishPost = 4;

	// the barcode symbology for Canada post
	static readonly SymbologyCanadaPost = 5;

	// the barcode symbology Chinese 2 of 5
	static readonly SymbologyChinese2of5 = 6;

	// the barcode symbology Codabar
	static readonly SymbologyCodabar = 7;

	// the barcode symbology Codablock A
	static readonly SymbologyCodablockA = 8;

	// the barcode symbology Codablock F
	static readonly SymbologyCodablockF = 9;

	// the barcode symbology Code 11
	static readonly SymbologyCode11 = 10;

	// the barcode symbology Code 39
	static readonly SymbologyCode39 = 11;

	// the barcode symbology Code 39 Extended
	static readonly SymbologyCode39Extended = 12;

	// the barcode symbology Code 39 Trioptic
	static readonly SymbologyCode39Trioptic = 13;

	// the barcode symbology Code 93
	static readonly SymbologyCode93 = 14;

	// the barcode symbology Code 128
	static readonly SymbologyCode128 = 15;

	// the barcode symbology DataMatrix
	static readonly SymbologyDataMatrix = 16;

	// the barcode symbology for Dutch post
	static readonly SymbologyDutchPost = 17;

	// the barcode symbology EAN 8
	static readonly SymbologyEan8 = 18;

	// the barcode symbology EAN 13
	static readonly SymbologyEan13 = 19;

	// the barcode symbology EAN 128
	static readonly SymbologyEan128 = 20;

	// the barcode symbology EAN 128 Irregular
	static readonly SymbologyEan128Irregular = 21;

	// the barcode symbology EAN UCC Composite AB
	static readonly SymbologyEanUccCompositeAB = 22;

	// the barcode symbology EAN UCC Composite C
	static readonly SymbologyEanUccCompositeC = 23;

	// the barcode symbology GS1 Databar
	static readonly SymbologyGs1Databar = 24;

	// the barcode symbology GS1 Databar Limited
	static readonly SymbologyGs1DatabarLimited = 25;

	// the barcode symbology GS1 Databar Expanded
	static readonly SymbologyGs1DatabarExpanded = 26;

	// the barcode symbology Interleaved 2 of 5
	static readonly SymbologyInterleaved2of5 = 27;

	// the barcode symbology ISBT 128
	static readonly SymbologyIsbt128 = 28;

	// the barcode symbology for Japan post
	static readonly SymbologyJapanPost = 29;

	// the barcode symbology Matrix 2 of 5
	static readonly SymbologyMatrix2of5 = 30;

	// the barcode symbology Maxi Code
	static readonly SymbologyMaxicode = 31;

	// the barcode symbology MSI
	static readonly SymbologyMsi = 32;

	// the barcode symbology PDF 417
	static readonly SymbologyPdf417 = 33;

	// the barcode symbology PDF 417 Micro
	static readonly SymbologyPdf417Micro = 34;

	// the barcode symbology Planet
	static readonly SymbologyPlanet = 35;

	// the barcode symbology Plessey
	static readonly SymbologyPlessey = 36;

	// the barcode symbology Postnet
	static readonly SymbologyPostnet = 37;

	// the barcode symbology QR Code
	static readonly SymbologyQRCode = 38;

	// the barcode symbology Standard 2 of 5
	static readonly SymbologyStandard2of5 = 39;

	// the barcode symbology Telepen
	static readonly SymbologyTelepen = 40;

	// the barcode symbology TLC 39
	static readonly SymbologyTlc39 = 41;

	// the barcode symbology UPC A
	static readonly SymbologyUpcA = 42;

	// the barcode symbology UPC E0
	static readonly SymbologyUpcE0 = 43;

	// the barcode symbology UPC E1
	static readonly SymbologyUpcE1 = 44;

	// the barcode symbology USPS Intelligent Mail
	static readonly SymbologyUspsIntelligentMail = 45;

	// the barcode symbology Direct Part Marking
	static readonly SymbologyDirectPartMarking = 46;

	// the barcode symbology Han Xin
	static readonly SymbologyHanXin = 47;

	// the barcode symbology DotCode
	static readonly SymbologyDotCode = 48;

	// the barcode symbology Digimarc
	static readonly SymbologyDigimarc = 49;

	// the last barcode symbology ID, not an actual barcode symbology
	static readonly LastSymbologyID = 50;

	// the RFID Tag Type ISO 14443 A
	static readonly TagTypeISO14443TypeA = 256;

	// the RFID Tag Type ISO 14443 B
	static readonly TagTypeISO14443TypeB = 257;

	// the RFID Tag Type Felica
	static readonly TagTypeFelica = 258;

	// the RFID Tag Type ISO 15693
	static readonly TagTypeISO15693 = 259;

	// the RFID Tag Type NXPI Code 1
	static readonly TagTypeNXPICODE1 = 260;

	// the RFID Tag Type Inside Secure Pico Tag
	static readonly TagTypeInsideSecurePicoTag = 261;

	// the RFID Tag Type Innovision Topaz Jewel
	static readonly TagTypeInnovisionTopazJewel = 262;

	// the RFID Tag Type Thin Film NFC Barcode
	static readonly TagTypeThinfilmNFCBarcode = 263;

	// the RFID Tag Type ST Micro Electronics SR
	static readonly TagTypeSTMicroElectronicsSR = 264;

	// the RFID Tag Type ASK CTS 256B or CTS 512B
	static readonly TagTypeASKCTS256BOrCTS512B = 265;

	// the RFID Tag Type NFC Forum
	static readonly TagTypeNFCForum = 266;

	// the RFID Tag Type Innovatron Radio Protocol
	static readonly TagTypeInnovatronRadioProtocol = 267;

	// the last RFID tag type, not an actual tag type
	static readonly TagTypeLastTagType = 268;

};


/**

 @brief define the data source status

 */
export class CaptureDataSourceStatus {
	// the data source status is disabled
	static readonly Disable = 0;

	// the data source status is enabled
	static readonly Enable = 1;

	// the data source is not supported
	static readonly NotSupported = 2;

};


/**

 @brief define the flag value of a data source structure

 */
export class CaptureDataSourceFlags {
	// the data source contains a status
	static readonly Status = 1;

	// the data source contains some parameters
	static readonly Param = 2;

};

